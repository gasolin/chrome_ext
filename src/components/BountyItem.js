import React from 'react';

const bountyImgStyle = {
  height: '20px',
  width: '20px'
};

export function BountyItem({item}) {
  return (
    <tr>
      <td><img src={item.imgsrc} alt='' style={bountyImgStyle} /></td>
      <td>{item.timeDiff}</td>
      <td>{item.tokenAmount}</td>
      <td>{item.title}</td>
      <td>{item.state}</td>
      <td><a href={item.linkUrl} target= '_blank' className='target' rel='noopener noreferrer'>View &gt;&gt;</a></td>
    </tr>
  )
}

export default BountyItem;
