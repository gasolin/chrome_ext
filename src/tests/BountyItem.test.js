import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import BountyItem from '../components/BountyItem';

const mockItem = {
  imgsrc: 'src="https://avatars1.githubusercontent.com/u/30044474?v=4',
  timeDiff: '1 month ago',
  tokenAmount: '0.06 ETH',
  title: 'Increase Code Coverage by 5%',
  state: 'open',
  linkUrl: 'href="https://github.com/gitcoinco/web/issues/408'
};

it('renders without crashing', () => {
  const tbody = document.createElement('tbody');
  ReactDOM.render(<BountyItem item={mockItem}/> , tbody);
  ReactDOM.unmountComponentAtNode(tbody);
});

it('render correctly', () => {
  const component = renderer.create(
    <table>
      <tbody>
        <BountyItem item={mockItem} />
      </tbody>
    </table>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
