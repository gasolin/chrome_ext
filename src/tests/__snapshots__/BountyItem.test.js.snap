// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        <img
          alt=""
          src="src=\\"https://avatars1.githubusercontent.com/u/30044474?v=4"
          style={
            Object {
              "height": "20px",
              "width": "20px",
            }
          }
        />
      </td>
      <td>
        1 month ago
      </td>
      <td>
        0.06 ETH
      </td>
      <td>
        Increase Code Coverage by 5%
      </td>
      <td>
        open
      </td>
      <td>
        <a
          className="target"
          href="href=\\"https://github.com/gitcoinco/web/issues/408"
          rel="noopener noreferrer"
          target="_blank"
        >
          View &gt;&gt;
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
